Formula/grafana-cloud-agent.rb

```diff
@@ -0,0 +1,72 @@
+class GrafanaCloudAgent < Formula
+  desc "Lightweight subset of Prometheus and more, optimized for Grafana Cloud"
+  homepage "https://grafana.com/products/cloud/"
+  url "https://github.com/grafana/agent/archive/v0.9.0.tar.gz"
+  sha256 "44d872044e0621a77e05f5608ba1bde68e39ddc2aaf0c1c64c53fa3cc7991aa1"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/agent"
```

Comment:
If I understand this correctly, this won't properly build the grafana cloud agent. We expect it to be built with `make` since we set some flags for version info and such: https://github.com/grafana/agent/blob/68dc22000bb219e6462c77470eea8eb9b7cee9a8/Makefile#L53

Reply:
More importantly, all your version info seemed to rely on `git`, so it's impossible to build cleanly from your release tarballs.

Reply:
Yes, the original PR used `make agent` rather than `go build`. I noticed the git errors too, but it was still possible to build the agent regardless of the errors so I pushed it through. I think upstream can eventually make it build cleanly when dealing with a non-git repo environment.

Reply:
It may be _possible_ to build, but:
```
$ brew install -s grafana-cloud-agent
...
$ grafana-cloud-agent -version
agent, version  (branch: , revision: )
  build user:       
  build date:       
  go version:       go1.15.5
  platform:         darwin/amd64
```
I'm pretty sure upstream would prefer that it not be released like this.

Reply:
(Looks like you already figured this out but I should mention I'm the creator and primary maintainer of the Grafana Cloud Agent, sorry for not making that clear.)

> @rfratto, More importantly, all your version info seemed to rely on git, so it's impossible to build cleanly from your release tarballs, even with make agent. That's something you might want to address.

It looks like this is a problem for most of the projects Grafana Labs makes, which is something we should fix. 

The fix I will make for this would likely still depend on being built with `make agent`, but would remove the hard dependency on being built with the git history intact. 

I'm comfortable with this being released as-is, since it's better for us to have presence in Homebrew with version quirks than no presence at all. Worst case scenario someone can run a `brew info` to see what version they installed. When we have a fix for this upstream, I (or @inf0rmer) can follow back up with a bump to the formula that uses `make agent` to build the binary to get the versioning info in.

Reply:
I don't think we should add new formulae with known issues.

Reply:
I completely understand that stance. I think given the current problems, we're likely going to put this in [Grafana Labs'](https://github.com/grafana/homebrew-grafana) tap first and follow up with an upstream PR to homebrew-core once we have the issues worked out. 

Thanks for taking the time to review this!